### ISSUE TITLE
Stacking multiple modals in parent element

### ISSUE THREAD

**Issue by edriang** [angular-ui/bootstrap#6044]
When I want to open multiple modals, new ones are put before old ones in DOM. This way, if I have a class that gives for example z-index 1000 to my modals, as all modals share same z-index, the ones that appears before in the DOM are covered by its modal siblings.

Looking at the code, I'm seeing that it is using $animate.enter to append the modal to the parent. This method has the following signature:
function enter(element, parent, after, options)

Currently, modal is using always "parent" attribute of "enter" function. But in this way, angular is internally prepending element in parten, NOT appending.

Probably the solution would be to check if there are some other modals opened in current parent. If they exists, then give reference to last modal in parent to append new modal after.

Angular: 1.4
UIBS: 1.3.2
Bootstrap: 3

**Comment by edriang** [comment]
Possible solution, in modal.js line 2660:

```
var after = appendToElement.find('[uib-modal-window]:last');
appendToElement.addClass(modalBodyClass);
$animate.enter($compile(angularDomEl)(modal.scope), appendToElement, after.length > 0 ? after : null);
```

**Comment by isometriq** [comment]
I have the same problem, when using "angular-bootstrap": "~2.0.0"

I was able to make it work by using the previous "angular-bootstrap": "~1.3.3" ..dunno if I'm missing a lot, but it works correctly that way.

My dependencies are...
```
{
    "angular": "~1.5",
    "angular-resource": "~1.5.8",
    "angular-sanitize": "~1.5.8",
    "angular-ui-router": "~0.2.0",
    "angular-bootstrap": "~1.3.3",
    "restangular": "~1.5.2",
    "satellizer": "~0.15.4",
    "ngstorage": "~0.3.10",
    "components-font-awesome": "~4.4.0",
    "bootswatch-dist": "united",
    "ngToast": "ngtoast#~2.0.0",
    "jquery": "~2.1.0"
}
```

**Comment by shinecita** [comment]
I have the same error now after upgrading from 1.3.3

**Comment by russell-miller** [comment]
Also having this problem. Smells like a bug that `appendTo` is prepending.

**Comment by shinecita** [comment]
I think its the z-index not properly set somewhere, will clone this to see

**Comment by edriang** [comment]
I solved the problem replacing lines 2661 and 2661 (vesion 1.3.2). 
"after" parameter was available but never used.

Old lines:
appendToElement.addClass(modalBodyClass);
$animate.enter($compile(angularDomEl)(modal.scope), appendToElement);

New lines:
var after = appendToElement.find('[uib-modal-window]:last');
appendToElement.addClass(modalBodyClass);
$animate.enter($compile(angularDomEl)(modal.scope), appendToElement, after.length > 0 && after);

I don't have cloned this repo, so if someone want to pull this modification is welcomed to do it.
This solution worked OK for me.

**Comment by shinecita** [comment]
if (modal.animation) {
          angularDomEl.attr('modal-animation', 'true');
        }

        modal.scope.index = topModalIndex; // ADDED To Patch index bug

        appendToElement.addClass(modalBodyClass);

        $animate.enter($compile(angularDomEl)(modal.scope), appendToElement);

**Comment by shinecita** [comment]
Oh, somebody just fixed in a new build.

**Comment by wesleycho** [comment]
Closing as addressed by #6104

**Comment by pbrain19** [comment]
Issue has reappeared in latest.
